syft/frameworks/torch/hook.py

```diff
@@ -464,19 +476,60 @@ def overloaded_pointer_method(self, *args, **kwargs):
 
             # Send the command
             command = (attr, self, args, kwargs)
+
             response = owner.send_command(location, command)
 
             return response
 
         return overloaded_pointer_method
 
+    def get_hooked_additive_shared_method(hook_self, attr):
```

Comment:
Is there a way that we can do this generically, It seems like there is alot of code rewriting here

Reply:
Hum not trivially I think, the thing on the shares reconstruction is typical to AdditiveSharedTensor and MultiPointerTensor